internal/remotestorage/remotestorage.go

```diff
@@ -0,0 +1,65 @@
+package remotestorage
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+// RemoteStorage defines functions to interact with application storage service
+type RemoteStorage interface {
+	Upload(url, fileName, formType string) ([]byte, error)
```

Comment:
1. Having the URL as part of the interface means that all implementations have to have a URL. Isn't it better to leave this up to the implementation?
2. It might be better to define a proper response type instead of `[]byte`. That way, the specific implementation can adhere to it. With just `[]byte` and potentially different implementations, how is the caller supposed to know how to deal with this response body?

Reply:
Let me update this a bit.